src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -189,10 +205,17 @@ class Decimal extends React.Component {
     }
 
     const withoutDelimiters = this.removeDelimiters(i18nValue);
-    const { separator } = I18nHelper.format();
+    const separator = this.getSeparator("decimal");
     return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
   };
 
+  getSeparator(separatorType) {
+    const numberWithGroupAndDecimalSeparator = 1000.1;
```

Comment:
This doesn't work for `es-ES` or `pt-PT`, these are the locales in the original issue, I think you need to change the value to `10000.1`
```suggestion
    const numberWithGroupAndDecimalSeparator = 10000.1;
```